spec/requests/api/provision_requests_spec.rb

```diff
@@ -105,6 +105,140 @@ def app
+
+    let(:expected_provreq_hash) do
+      {
+        "userid"         => api_config(:user),
+        "requester_name" => api_config(:user_name),
+        "approval_state" => "pending_approval",
+        "type"           => "MiqProvisionRequest",
+        "request_type"   => "template",
+        "message"        => /Provisioning/i,
+        "status"         => "Ok"
+      }
+    end
+
+    it "supports manual placement" do
+      api_basic_authorize collection_action_identifier(:provision_requests, :create)
+
+      # dialog  # Create the Provisioning dialog
+      aws_dialog # Create the AWS Provisioning dialog
+      run_post(provision_requests_url, provreq1_body)
+
+      # pp @result
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_provreq_attributes)
+      expect_results_to_match_hash("results", [expected_provreq_hash])
+
+      options = @result["results"].first["options"]
+      # pp options
+      expect(options["placement_auto"]).to eq([false, 0])
+      expect(options["placement_availability_zone"]).to eq(['xxx'])
+      expect(options["cloud_network"]).to eq(['xxx'])
+      expect(options["cloud_subnet"]).to eq(['xxx'])
+      expect(options["security_groups"]).to eq(['xxx'])
+      expect(options["floating_ip_address"]).to eq(['xxx'])
+
+      task_id = @result["results"].first["id"]
+      expect(MiqProvisionRequest.exists?(task_id)).to be_truthy
+    end
+
+    it "does not process manual placement data if placement_auto is not set" do
+      api_basic_authorize collection_action_identifier(:provision_requests, :create)
+
+      # dialog  # Create the Provisioning dialog
+      aws_dialog # Create the AWS Provisioning dialog
+      run_post(provision_requests_url, provreq2_body)
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_provreq_attributes)
+      expect_results_to_match_hash("results", [expected_provreq_hash])
+
+      options = @result["results"].first["options"]
+      expect(options["placement_auto"]).to eq([true, 1])
+      expect(options["placement_availability_zone"]).to be_nil
+
+      task_id = @result["results"].first["id"]
+      expect(MiqProvisionRequest.exists?(task_id)).to be_truthy
+    end
+
+    it "does not process manual placement data if placement_auto is set to true" do
+      api_basic_authorize collection_action_identifier(:provision_requests, :create)
+
+      # dialog  # Create the Provisioning dialog
+      aws_dialog # Create the AWS Provisioning dialog
+      run_post(provision_requests_url, provreq3_body)
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_provreq_attributes)
+      expect_results_to_match_hash("results", [expected_provreq_hash])
+
+      options = @result["results"].first["options"]
+      expect(options["placement_auto"]).to eq([true, 1])
+      expect(options["placement_availability_zone"]).to be_nil
```

Comment:
The stored value is an array.  Suggest using:`expect(options["placement_availability_zone"].first).to eq nil`